presto-main/src/main/resources/webapp/src/components/ResourceGroupView.jsx

```diff
@@ -0,0 +1,349 @@
+
+const SPARKLINE_PROPERTIES = Object.freeze({
+    disableHiddenCheck: true,
+    fillColor: '#3F4552',
+    height: '75px',
+    lineColor: '#747F96',
+    spotColor: '#1EDCFF',
+    tooltipClassname: 'sparkline-tooltip',
+    width: '100%',
+});
+
+// One information entry of the resource group
+function InfoItem({name, value}) {
+    return (
+        <tr>
+            <td className="info-title">
+                {name}
+            </td>
+            <td className="info-text wrap-text">
+                {value}
+            </td>
+        </tr>
+    );
+}
+
+// One timeline chart of the resource group
+function TimelineItem({name, chartId, value}) {
+    return (
+        <>
+        <tr>
+            <td className="info-title">
+                {name}
+            </td>
+            <td rowSpan="2">
+                <div className="query-stats-sparkline-container">
+                    <span className="sparkline" id={chartId}><div className="loader">Loading ...</div></span>
+                </div>
+            </td>
+        </tr>
+        <tr className="tr-noborder">
+            <td className="info-sparkline-text">
+                {value}
+            </td>
+        </tr>
+        </>
+    );
+}
+
+// Display the root group list if no group is specified or general information if any error.
+function NoGroupIdWidget({groupId, error, groups}) {
+    let docView = (<h3 className="text-center">{ groupId ? 'Retrieving resource group information...' : 'Detecting Resource Groups settings...'}</h3>);
+
+    if (groups.length > 0) {
+        docView = (
+            <>
+            <div className="col-xs-4 col-xs-offset-4">
+                <h4 className="text-center">Available resource groups:</h4>
+                <div className="list-group">
+                    {groups.map(grp => (
+                    <a className="list-group-item text-center" href={'./res_groups.html?group=' + encodeURIComponent(grp.id[0])}>{truncateString(grp.id[0], 35)}</a>
+                    ))}
+                </div>
+            </div>
+            </>
+        );
+    } else if (error) {
+        docView = (
+            <div className="col-xs-12">
+                <h3 className="text-center">For details information about Resource Groups, please check document <a href="https://prestodb.io/docs/current/admin/resource-groups.html">here</a></h3>
+                <h4 className="text-center">{error.message}</h4>
+            </div>
+        )
+    }
+
+    return (
+        <div className="col-xs-12">{docView}</div>
+    );
+}
+
+function GroupBreadcrumb({groupId}) {
+    let fullResPath = '';
+    const lastIdx = groupId.length - 1;
+    const path = groupId.map((rs, idx) => {
+        if (idx === 0) {
+            fullResPath = rs;
+        } else {
+            fullResPath = fullResPath + '.' + rs;
+        }
+        if (idx === lastIdx) {
+            return (
+                <li key={'_' + idx} className="active">{rs}</li>
```

Comment:
Consider removing the underscore prefix (`_`) here and in similar instances.

```suggestion
                <li key={idx} className="active">{rs}</li>
```